modules/persistence/src/services/metadata/associated_products/index.ts

```diff
@@ -52,54 +61,65 @@ const getAllAssociatedRepoBranchesEntries = async (metadata: SharedMetadata) =>
 const mapRepoBranches = (repoBranches: ReposBranchesDocument[]) =>
   Object.fromEntries(
     repoBranches.map((entry) => {
-      const branches = Object.fromEntries(entry.branches.map((branch) => [branch.gitBranchName, branch]));
-      return [entry.project, branches];
+      const { url, prefix } = entry;
+      const branches = Object.fromEntries(entry.branches.map((branch) => [branch.gitBranchName, { ...branch }]));
+      return [entry.project, { ...branches, url, prefix }];
     })
   );
 
-const hasAssociations = (metadata) => !!metadata.associated_products?.length;
+export const hasAssociations = (metadata) => !!metadata.associated_products?.length;
 
-// TODO: update param to target associated project name.
-// confusing - takes in associated metadata and returns umbrella metadata
-const umbrellaMetadataEntry = async (metadata): Promise<SharedMetadata> => {
+const umbrellaMetadataEntry = async (project: string): Promise<Metadata> => {
   try {
     const snooty = await db();
     const entry = await snooty
       .collection('metadata')
       .find({
-        'associated_products.name': metadata.project,
+        'associated_products.name': project,
       })
       .sort({ build_id: -1 })
       .limit(1)
       .toArray();
-    return entry[0] as unknown as SharedMetadata;
+    return entry[0] as unknown as Metadata;
   } catch (error) {
     console.log(`Error at time of querying for umbrella metadata entry: ${error}`);
     throw error;
   }
 };
 
-// Convert our cursor from the shared ToC aggregation query into a series of ToC objects
+// Convert our cursor from the associated metadata aggregation query into a series of ToC objects and their parent metadata entries
 const shapeToCsCursor = async (
   tocCursor: AggregationCursor,
   repoBranchesMap
-): Promise<{ tocInsertions: ToCInsertions; tocOrderInsertions: TocOrderInsertions }> => {
+): Promise<{
+  tocInsertions: ToCInsertions;
+  tocOrderInsertions: TocOrderInsertions;
+  associatedMetadataEntries: Metadata[];
+}> => {
   const tocInsertions = {};
   const tocOrderInsertions = {};
+  const associatedMetadataEntries: Metadata[] = [];
 
   await tocCursor.forEach((doc) => {
     // Initialize to empty object if we haven't already, for a given project.
     if (!tocInsertions[doc._id.project]) tocInsertions[doc._id.project] = {};
     if (!tocOrderInsertions[doc._id.project]) tocOrderInsertions[doc._id.project] = {};
 
+    const repoBranchesEntry = repoBranchesMap?.[doc._id.project]?.[doc._id.branch];
     // TODO: If we want staging builds with embedded versions, it needs to be added here
-    if (repoBranchesMap?.[doc._id.project]?.[doc._id.branch]) {
-      tocInsertions[doc._id.project][doc._id.branch] = doc.most_recent.toctree;
+    if (repoBranchesEntry) {
+      const { url, prefix } = prefixFromEnvironment(repoBranchesEntry);
```

Comment:
discussed offline, leaving here for note. passing in `BranchEntry` here when expecting `ReposBranchesDocument`. can be optimized to `projectInfo` vs `branchInfo`